deploy_tools/deploy.py

```diff
@@ -77,7 +77,7 @@ def wait_for_successful_transaction_receipt(web3: Web3, txid: str, timeout=180)
     """
     receipt = web3.eth.waitForTransactionReceipt(txid, timeout=timeout)
     status = receipt.get("status", None)
-    if status is False:
+    if status == 0 or status is False:
```

Comment:
I am thinking that it would be maybe safer to check the opposite, as if we would have done this, we would not had this bug. 
```if not (status == 1 or status is True)```

Reply:
if we would have done `if not status is True` we would have always raised.

Reply:
@schmir Opinion?

Reply:
1. I think you should not compare with `is True` or `is False`. This is kind of awkward and I do think at least some of the linters will warn about that.
2. In the comparison `status ==1 or status is True` and `status == 0 or status is False` the second condition will never be true, because `True` will compare equal to 1 and `False` will compare equal to 0, i.e. the comparisons are equal to `status == 1` and `status==0`

Maybe we should raise an exception, when we get a value, which we don't know about.
So, it should be something like:
```
if status == 0:   # include status==False
   ...
elif status == 1: # includes status == True
  ...
else: 
  raise 
```

Reply:
>     2\. In the comparison `status ==1 or status is True` and `status == 0 or status is False` the second condition will never be true,

I will not be evaluated (because the first condition is already true) or it will always be false.

Reply:
> I think you should not compare with is True or is False. This is kind of awkward and I do think at least some of the linters will warn about that.

I believe it is the opposite, I cannot use `status == False` and flake8 wants `status is False`. I also think this rule is odd and black does not fix it automatically so if we can disable it I'd be happy.

For the rest I agree.

Reply:
> I believe it is the opposite, I cannot use `status == False` and flake8 wants `status is False`. I also think this rule is odd and black does not fix it automatically so if we can disable it I'd be happy.

flake8 says it wants the following:
```
E712 comparison to True should be 'if cond is True:' or 'if cond:'
```
I consider the `is True`/`is False` bad style. 

Anyway, what I suggessted is that we compare against 0 and 1.

You could also use something like `if status in (True, 1): ...` if you want to make it explicit that it also matches `True`.

Reply:
Fine with me